sklearn/tree/tests/test_export.py

```diff
@@ -234,3 +244,52 @@ def test_friedman_mse_in_graphviz():
 
     for finding in finditer("\[.*?samples.*?\]", dot_data.getvalue()):
         assert_in("friedman_mse", finding.group())
+
+
+def test_decimals():
+    # regression case
+    rng = RandomState(2)
+    X = rng.random_sample((5, 2))
+    y_reg = rng.random_sample((5, ))
+
+    clf = DecisionTreeRegressor(criterion="friedman_mse", random_state=0,
+                                max_depth=1)
+    clf.fit(X, y_reg)
+    for decimals in (4, 3):
+        dot_data = StringIO()
+        export_graphviz(clf, out_file=dot_data, decimals=decimals)
+        # check value
+        for finding in finditer("value = \d+\.\d+", dot_data.getvalue()):
+            assert_equal(len(search("\.\d+", finding.group()).group()),
+                         decimals + 1)
+        # check impurity
+        for finding in finditer("friedman_mse = \d+\.\d+",
+                                dot_data.getvalue()):
+            assert_equal(len(search("\.\d+", finding.group()).group()),
```

Comment:
I agree that `assert_less` makes this more permissive. In that case, this warrants a comment that this is particular for this random seed as you mention or you can use toy data where you know that the number of decimals in `impurity` are greater than the provided value.